controllers/ldap.go

```diff
@@ -409,9 +418,17 @@ func (c *ApiController) SyncLdapUsers() {
 		return
 	}
 
-	exist, failed, err := object.SyncLdapUsers(goCtx, owner, users, ldapId)
+	syncResult, err := object.SyncLdapUsers(goCtx, object.LdapSyncCommand{
+		Owner:     owner,
+		LdapUsers: users,
+		LdapId:    ldapId,
+		Reason:    "manual",
+	})
 	if err != nil {
 		record.AddReason(fmt.Sprintf("LDAP error: %s", err.Error()))
+
+		c.ResponseError(err.Error())
+		return
 	}
 
 	err = object.UpdateLdapSyncTime(ldapId)
```

Comment:
since object.SyncLdapUsers could create new users via `object.AddUser`
Can we use trx here for 

```
SyncLdapUsers
UpdateLdapSyncTime(ldapId)
```

Reply:
Can you provide an example of how that is done in casgate?

Reply:
check 
`type TransactionManager interface `